### PR TITLE
boards: removed support for pca1000[0|5]

### PR DESCRIPTION
**Pull request by haukepetersen**
Both boards were early development kits by Nordic and are to my knowledge discontinued, as both have (way more affordable) successors:
- `pca10000` -> `nrf51dongle`
- `pca10005` -> `nrf51dk`

Also we don't have a `pca10000` available to us anymore (went up in smoke quite a while ago), and the `pca10005` is not trivial to flash (needs external jlink adapter...), so we have not tested it in probably 2 or more years. Further it seems that there are not many of these boards around in general, so I don't think someone will even notice that we drop their support...

Will provide entries to the `LOSTANDFOUND.md` in a separate PR.
EDIT: see #7980

### CODE REVIEW

**Comment by gebart** [comment]
:+1:

**Comment by haukepetersen** [comment]
I think I forgot to look for blacklistings throughout the tests, will do that now.

**Comment by haukepetersen** [comment]
ok, good thing we talked about it :-) Removed all occurences of pca1000x from the examples and applications.

**Comment by kaspar030** [comment]
&go.

File: examples/gnrc_border_router/Makefile [hunk 1]
```diff
@@ -12,8 +12,8 @@ BOARD_INSUFFICIENT_MEMORY := airfy-beacon b-l072z-lrwan1 bluepill calliope-mini
                              microbit msb-430 msb-430h nrf51dongle nrf6310 \
                              nucleo32-f031 nucleo32-f042 nucleo32-f303 nucleo32-l031 \
                              nucleo-f030 nucleo-f070 nucleo-f072 nucleo-f103 nucleo-f302 \
-                             nucleo-f334 nucleo-l053 nucleo-l073 opencm904 pca10000 \
-                             pca10005 spark-core stm32f0discovery telosb wsn430-v1_3b \
+                             nucleo-f334 nucleo-l053 nucleo-l073 opencm904 \
+                             spark-core stm32f0discovery telosb wsn430-v1_3b \
```
Comment (miri64) [inline examples/gnrc_border_router/Makefile]: This line cause a merge conflict in the gnrc_netif2 integration (see #7925). Is there a way to revert this for now?

Comment (kaspar030) [reply examples/gnrc_border_router/Makefile]: Isn't that only relevant when you actually merge?

Comment (haukepetersen) [reply examples/gnrc_border_router/Makefile]: how about #7986

**Pull request merged**